OPAL/br/src/main/scala/org/opalj/br/fpcf/analyses/FieldAccessInformationAnalysis.scala

```diff
@@ -0,0 +1,152 @@
+                                OPALLogger.warn("project configuration", message)
+                            }
+                        // IMPROVE: In the old implementation, unresolvable field accesses were tracked. This can
+                        // also be done here but is not needed as of now.
+                    }
+
+                case PUTFIELD.opcode | PUTSTATIC.opcode =>
+                    val fieldWriteAccess = instruction.asInstanceOf[FieldWriteAccess]
+                    resolveFieldReference(fieldWriteAccess) match {
+                        case Some(field) =>
+                            writeAccesses.getOrElseUpdate(field, new IntTrieSetBuilder()) += pc
+                        case None =>
+                            if (reportedFieldAccesses.add(instruction)) {
+                                val message = s"cannot resolve field write access: $instruction"
+                                OPALLogger.warn("project configuration", message)
+                            }
+                        // IMPROVE: In the old implementation, unresolvable field accesses were tracked. This can
+                        // also be done here but is not needed as of now.
+                    }
+
+                case _ => /*nothing to do*/
+            }
+        }
+
+        val definedMethod = declaredMethods(method)
+        val fieldAccesses = mutable.AnyRefMap.empty[Field, (List[(DefinedMethod, PCs)], List[(DefinedMethod, PCs)])]
+        readAccesses foreach { readAccess =>
+            val (field, pcs) = readAccess
+            fieldAccesses.get(field) match {
+                case Some(currentAccesses) =>
+                    fieldAccesses.put(field, ((definedMethod, pcs.result()) :: currentAccesses._1, currentAccesses._2))
+                case None =>
+                    fieldAccesses.put(field, ((definedMethod, pcs.result()) :: Nil, Nil))
+            }
+        }
+        writeAccesses foreach { writeAccess =>
+            val (field, pcs) = writeAccess
+            fieldAccesses.get(field) match {
+                case Some(currentAccesses) =>
+                    fieldAccesses.put(field, (currentAccesses._1, (definedMethod, pcs.result()) :: currentAccesses._2))
+                case None =>
+                    fieldAccesses.put(field, (Nil, (definedMethod, pcs.result()) :: Nil))
+            }
+        }
+
+        Results(fieldAccesses map { fieldAccess =>
+            val readAccesses = fieldAccess._2._1.to(Set)
+            val writeAccesses = fieldAccess._2._2.to(Set)
+
+            new PartialResult(
+                fieldAccess._1,
+                FieldAccessInformation.key,
+                (current: EOptionP[Field, FieldAccessInformation]) => current match {
+                    case InterimUBP(ub: FieldAccessInformation) =>
+                        val newFAI = FieldAccessInformation(readAccesses, writeAccesses)
+
+                        if (ub == newFAI)
```

Comment:
Can this ever be the case? Seems to be a rather expensive check for something that will probably never succeed?